src/functions.jl

```diff
@@ -5,33 +5,22 @@
 
 Inverse of `sqrt(x)` for non-negative `x`.
 """
-square(x) = x^2
-function square(x::Real)
-    x < zero(x) && throw(DomainError(x, "`square` is defined as the inverse of `sqrt` and can only be evaluated for non-negative values"))
+function square(x)
+    if is_real_type(typeof(x)) && x < zero(x)
+        throw(DomainError(x, "`square` is defined as the inverse of `sqrt` and can only be evaluated for non-negative values"))
+    end
     return x^2
 end
 
 
-function invpow2(x::Real, p::Integer)
-    if x ≥ zero(x) || isodd(p)
-        copysign(abs(x)^inv(p), x)
+function invpow2(x::Number, p::Real)
```

Comment:
We should probably name this `invpow` instead of `invpow2`, right?

Reply:
Ah, right, that was the reason. Well, these are still internal, so we can rename them - maybe `invpov_arg2`?